public/app/features/alerting/unified/utils/rule-form.ts

```diff
@@ -599,6 +602,73 @@ export const panelToRuleFormValues = async (
   return formValues;
 };
 
+export const scenesPanelToRuleFormValues = async (
+  vizPanel: VizPanel,
+  scenesQueries: SceneQueryRunner,
+  scenesDashboard: DashboardScene
+): Promise<Partial<RuleFormValues> | undefined> => {
+  if (!vizPanel.state.key || !scenesDashboard.state.uid) {
+    return undefined;
+  }
```

Comment:
Maybe we can simplify this a bit, and `ScenesNewRuleFromPanelButton` props, so that it accepts only VizPanel?

Dashboard for a viz panel can be resolved using `getDashboardSceneFor` util and the query runner using `getQueryRunnerFor`. It's a nit but will allow us to change the structure in the future and rely on these predefined functions that resolve particular objects for a panel.

Reply:
Ah, good idea.